persistence/src/main/java/com/tapptitude/persistence/di/DatabaseModule.kt

```diff
@@ -0,0 +1,31 @@
+package com.tapptitude.persistence.di
+
+import android.app.Application
+import androidx.room.Room
+import androidx.room.migration.Migration
+import androidx.sqlite.db.SupportSQLiteDatabase
+import com.tapptitude.persistence.LocalDatabase
+import com.tapptitude.persistence.images.ImagesDao
+import org.koin.android.ext.koin.androidApplication
+import org.koin.dsl.module
+
+private const val DATABASE_NAME = "LocalStore.db"
+
+val databaseModule = module {
+    fun provideDatabase(application: Application): LocalDatabase =
```

Comment:
@greenfodor I would extract the effective creation of objects into a factory class so that we can keep the module files clean with only the Koin related logic. Check `NetworkComponentsFactory` for sample. 

Same goes for the `Migration`s part, it would be nicer to have them separately defined, so we could easier keep track when time passes, otherwise we will have a really long `addMigrations` block.

It looks much more clean to me. Open to discussion.